src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -0,0 +1,627 @@
+            }
+
+            long[] deltas = new long[]{tsDelta, tDelta, sizeDelta};
+            if (interArrival.computeDeltas(timestamp, arrivalTimeMs,
+                    (int) payloadSize, deltas, nowMs)) {
+                double tsDeltaMs = (1000.0 * tsDelta)
+                        / (1 << kInterArrivalShift);
+                estimator.update(tDelta, tsDeltaMs, sizeDelta,
+                        detector.getState(),
+                        arrivalTimeMs);
+                detector.detect(estimator.getOffset(), tsDeltaMs,
+                        estimator.getNumOfDeltas(), arrivalTimeMs);
+            }
+            if (!updateEstimate) {
+                // Check if it's time for a periodic update or if we
+                // should update because of an over-use.
+                if (lastUpdateMs == -1 ||
+                        nowMs - lastUpdateMs > remoteRate
+                                .getFeedBackInterval())
+                {
+                    updateEstimate = true;
+                }
+                else if (detector.getState() == BandwidthUsage.kBwOverusing)
+                {
+                    long incomingRate_ =
+                            incomingBitrate.getRate(arrivalTimeMs);
+                    if (incomingRate_ > 0 &&
+                            remoteRate.isTimeToReduceFurther(nowMs,
+                                    incomingBitrate_))
+                    {
+                        updateEstimate = true;
+                    }
+                }
+            }
+            if (updateEstimate) {
+                // The first overuse should immediately trigger a new estimate.
+                // We also have to update the estimate immediately if we are
+                // overusing and the target bitrate is too high compared to
+                // what we are receiving.
+                RateControlInput input = new RateControlInput(detector
+                        .getState(), incomingBitrate.getRate(arrivalTimeMs),
+                        estimator.getVarNoise());
+
+                /**
+                 * @Todo remote.update returns nothing. This is fine since we
+                 * can retrieve the value from remoteRate.getLatestEstimate
+                 */
+                remoteRate.update(input, nowMs);
+                targetBitrateBps = remoteRate.getLatestEstimate();
+                updateEstimate = remoteRate.isValidEstimate();
+                ssrcs = Keys(ssrcs_);
+            }
+        }
+        if (updateEstimate) {
+            lastUpdateMs = nowMs;
+            /**
+             * @Todo Fix Problem with transforming Long to Integer
+             */
+            observer_.onReceiveBitrateChanged(null, targetBitrateBps);
+        }
+    }
+
+    void Process() {}
+
+    long timeUntilNextProcess()
+    {
+        long kDisabledModuleTime = 1000;
+        return kDisabledModuleTime;
+    }
+
+    void timeoutStreams(long nowMs)
+    {
+        synchronized (critSect) {
```

Comment:
see line 114 of [webrtc absSendTime.h](https://chromium.googlesource.com/external/webrtc/+/master/webrtc/modules/remote_bitrate_estimator/remote_bitrate_estimator_abs_send_time.h)